--- src/compat.h.orig	2026-09-08 08:28:54.436539500 +0000
+++ src/compat.h	2026-09-08 08:29:03.794666840 +0000
@@ -61,6 +61,12 @@
 #include <getopt.h>
 #endif
 
+#ifdef _WIN32
+#include <direct.h>
+#define mkdir(p,m) _mkdir(p)
+#define fsync(fd) _commit(fd)
+#endif
+
 #ifdef HAVE_IO_H
 #include <io.h>
 #endif
--- src/compat.c.orig	2026-09-08 08:28:54.438290025 +0000
+++ src/compat.c	2026-09-08 08:28:54.456167907 +0000
@@ -9,7 +9,9 @@
 #include <errno.h>
 #include <assert.h>
 #include <locale.h>
+#ifndef _WIN32
 #include <langinfo.h>
+#endif
 
 /*
     Useful string stuff
@@ -199,7 +201,11 @@
   {
 #ifdef HAVE_ICONV
     setlocale(LC_ALL, "");
+    #ifdef _WIN32
+    strncpy(default_charset_buf, "UTF-8", sizeof default_charset_buf - 1);
+#else
     strncpy(default_charset_buf, nl_langinfo(CODESET), sizeof default_charset_buf - 1);
+#endif
     default_charset_buf[sizeof default_charset_buf - 1] = 0;
     default_charset = default_charset_buf;
     // fprintf(stderr, "INFO: default codeset is \"%s\"\n", default_charset); /* debug */
--- src/subgen.c.orig	2026-09-16 11:57:47.288546908 +0000
+++ src/subgen.c	2026-09-16 11:57:47.290372361 +0000
@@ -33,7 +33,11 @@
 #include <errno.h>
 #include <fcntl.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
+#endif
 
 #include "common.h"
 #include "conffile.h"
--- src/subfont.c.orig	2026-09-16 11:57:47.292256044 +0000
+++ src/subfont.c	2026-09-16 11:58:01.587913948 +0000
@@ -44,7 +44,11 @@
 
 #include <math.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
+#endif
 
 #include "subglobals.h"
 #include "subfont.h"
@@ -135,7 +139,6 @@
     if ((homedir = getenv("HOME")) == NULL)
 #if defined(__MINGW32__) || defined(__CYGWIN__) /*hack to get fonts etc. loaded outside of cygwin environment*/
       {
-        extern int __stdcall GetModuleFileNameA(void* hModule, char* lpFilename, int nSize);
         int i, imax = 0;
         GetModuleFileNameA(NULL, exedir, sizeof exedir);
         for (i = 0; i<   strlen(exedir); i++)
